src/Executables/Benchmark/Benchmark.cpp

```diff
@@ -0,0 +1,106 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <benchmark/benchmark.h>
+#include <vector>
+
+#include "DataStructures/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Numerical/Spectral/LegendreGaussLobatto.hpp"
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Element.hpp"
+#include "PointwiseFunctions/MathFunctions/PowX.hpp"
+// #include "Functors/Gradient.hpp"
+#include "Domain/GridCoordinates.hpp"
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+// This file is an example of how to do microbenchmark with Google Benchmark
+// https://github.com/google/benchmark
+// For two examples in different anonymous namespaces
+
+namespace {
+// Benchmark of push_back() in std::vector, following Chandler Carruth's talk
+// at CppCon in 2015,
+// https://www.youtube.com/watch?v=nXaxk27zwlk&feature=youtu.be
+
+static void bench_create(benchmark::State &state) {
+ while (state.KeepRunning()) {
+   std::vector<int> v;
+   benchmark::DoNotOptimize(&v);
+   static_cast<void>(v);
+ }
+}
+BENCHMARK(bench_create);
+
+static void bench_reserve(benchmark::State &state) {
+ while (state.KeepRunning()) {
+   std::vector<int> v;
+   v.reserve(1);
+   benchmark::DoNotOptimize(v.data());
+ }
+}
+BENCHMARK(bench_reserve);
+
+static void bench_push_back(benchmark::State &state) {
+ while (state.KeepRunning()) {
+   std::vector<int> v;
+   v.reserve(1);
+   benchmark::DoNotOptimize(v.data());
+   v.push_back(42);
+   benchmark::ClobberMemory();
+ }
+}
+BENCHMARK(bench_push_back);
+}  // namespace
+
+namespace {
+// In this anonymous namespace is an example of microbenchmarking the
+// all_gradient routine for the GH system
+
+template <size_t Dim>
+struct Kappa : db::DataBoxTag {
+  using type = tnsr::abb<DataVector, Dim, Frame::Grid>;
+  static constexpr db::DataBoxString_t label = "Kappa";
+};
+template <size_t Dim>
+struct psi : db::DataBoxTag {
```

Comment:
psi -> Psi

Reply:
This is an interesting topic we should sort out. In papers it is lower case psi, so what do we want to do in code?

Reply:
I think if we have a style for how structs should be cased we should stick to that regardless of the casing of the greek letters we use. E.g. if we're going to use the tag UpperSpatialMetric we should use Psi. If we decided to snake_case tags instead then Kappa above should be lowercased, but if we do that then using `psi` as a variable would have the same casing as its tag which I could see being confusing.

Reply:
We could replace psi with SpacetimeMetric, which avoids any reliance on specific naming conventions.  We may still end up with quantities that don't have common names (kappa, possibly) so we'd still need a decision there.  (Also do we like Tags::d and Tags::dt?)

Reply:
With the new Tensor things like kappa should go away, and so then we could use something like DtSpacetimeMetric and DerivSpacetimeMetric. Thoughts?

Personally I'm okay with Tags::dt, maybe Tags::d should be Tags::deriv?